test/k8sT/Bandwidth.go

```diff
@@ -0,0 +1,170 @@
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("K8sBandwidthTest", func() {
+	const (
+		testDS10   = "run=netperf-10"
+		testDS25   = "run=netperf-25"
+		testDS50   = "run=netperf-50"
+		testDSInf  = "run=netperf-inf"
+		testClient = "run=netperf-client"
+
+		maxRateDeviation = 4
+	)
+
+	var (
+		kubectl        *helpers.Kubectl
+		ciliumFilename string
+
+		backgroundCancel       context.CancelFunc = func() {}
+		backgroundError        error
+		enableBackgroundReport = true
+
+		podLabels = []string{
+			testDS10,
+			testDS25,
+			testDS50,
+			testDSInf,
+		}
+	)
+
+	BeforeAll(func() {
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+		ciliumFilename = helpers.TimestampFilename("cilium.yaml")
+		DeployCiliumAndDNS(kubectl, ciliumFilename)
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport(helpers.CiliumNamespace,
+			"cilium bpf bandwidth list",
+			"cilium endpoint list")
+	})
+
+	JustBeforeEach(func() {
+		if enableBackgroundReport {
+			backgroundCancel, backgroundError = kubectl.BackgroundReport("uptime")
+			Expect(backgroundError).To(BeNil(), "Cannot start background report process")
+		}
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+		backgroundCancel()
+	})
+
+	AfterEach(func() {
+		ExpectAllPodsTerminated(kubectl)
+	})
+
+	AfterAll(func() {
+		UninstallCiliumFromManifest(kubectl, ciliumFilename)
+		kubectl.CloseSSHClient()
+	})
+
+	SkipContextIf(func() bool {
+		return !helpers.RunsOnNetNextKernel()
+	}, "Checks Bandwidth Rate-Limiting", func() {
+		var demoYAML string
+
+		BeforeAll(func() {
+			demoYAML = helpers.ManifestGet(kubectl.BasePath(), "demo_bw.yaml")
+
+			res := kubectl.ApplyDefault(demoYAML)
+			res.ExpectSuccess("unable to apply %s", demoYAML)
+
+			podLabels := []string{
+				testDS10,
+				testDS25,
+				testDS50,
+				testDSInf,
+				testClient,
+			}
+			for _, label := range podLabels {
+				err := kubectl.WaitforPods(helpers.DefaultNamespace,
+					fmt.Sprintf("-l %s", label), helpers.HelperTimeout)
+				Expect(err).Should(BeNil())
+			}
+		})
+
+		AfterAll(func() {
+			_ = kubectl.Delete(demoYAML)
+		})
+
+		testNetperfFromPods := func(clientPodLabel, targetIP string, maxSessions, rate int) {
+			pods, err := kubectl.GetPodNames(helpers.DefaultNamespace, clientPodLabel)
+			ExpectWithOffset(1, err).Should(BeNil(), "cannot retrieve pod names by filter %q",
+				clientPodLabel)
+			for i := 1; i <= maxSessions; i++ {
+				cmd := helpers.SuperNetperf(i, targetIP, helpers.TCP_MAERTS, "")
+				for _, pod := range pods {
+					By("Running %d netperf sessions from %s pod to pod with IP %s (expected rate: %d)",
+						i, pod, targetIP, rate)
+					res := kubectl.ExecPodCmd(helpers.DefaultNamespace, pod, cmd)
+					ExpectWithOffset(1, res).Should(helpers.CMDSuccess(),
+						"Request from %s pod to pod with IP %s failed", pod, targetIP)
+					if rate > 0 {
+						Expect(res.InRange(rate, maxRateDeviation)).To(BeNil(), "Rate mismatch")
```

Comment:
This should be `ExpectWithOffset` to be consistent with the rest of the expectations in this function.